### PR TITLE
113 - Integrate Backend with Views

### PR DESCRIPTION
**Pull request by codyseibert**

### CODE REVIEW

File: shared/src/business/useCases/createCase.interactor.js [hunk 1]
```diff
@@ -34,6 +34,9 @@ exports.createCase = async ({ userId, documents, applicationContext }) => {
         ],
         docketNumber,
         documents,
+        caseTitle: `${
```
Comment (ericsorenson) [inline shared/src/business/useCases/createCase.interactor.js:37]: 👍

File: web-client/src/presenter/sequences/gotoDocumentDetailSequence.js [hunk 2]
```diff
@@ -1,3 +1,13 @@
+import clearAlerts from '../actions/clearAlertsAction';
+import getCase from '../actions/getCaseAction';
+import setCase from '../actions/setCaseAction';
+import setDocumentId from '../actions/setDocumentIdAction';
```
Comment (ericsorenson) [inline web-client/src/presenter/sequences/gotoDocumentDetailSequence.js:4]: You even alphabetized them :-)

File: web-client/src/tests/journey/docketClerkViewsDashboard.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  return it('Docket clerk views dashboard', async () => {
+    await test.runSequence('gotoDashboardSequence');
+    expect(test.getState('currentPage')).toEqual('DashboardDocketClerk');
+    expect(test.getState('workQueue').length).toBeGreaterThan(0);
+    const workItem = test
+      .getState('workQueue')
+      .find(workItem => workItem.docketNumber === test.docketNumber);
+    expect(workItem).toBeDefined();
+    test.documentId = workItem.document.documentId;
+    test.workItemId = workItem.workItemId;
```
Comment (ericsorenson) [inline web-client/src/tests/journey/docketClerkViewsDashboard.js:11]: 👍

File: web-client/src/views/WorkQueue.jsx [hunk 4]
```diff
@@ -50,7 +52,7 @@ export default connect(
                 </td>
                 <td>
                   <span className="responsive-label">Received</span>
-                  {item.messages[0].receivedDate || 'Dec 31, 1969'}
+                  {item.messages[0].createdAtFormatted || 'Dec 31, 1969'}
```
Comment (ericsorenson) [inline web-client/src/views/WorkQueue.jsx]: Is there a reason to display this default date?

**Review by dknesek — APPROVED** [review]
Really nice work, Cody. I have some questions/concerns about our entities overall that we can talk about after you merge this. I think what you have here is much better than when we were just cobbling together JSON.

But there are still some cases where we are initializing Entities with attributes that we shouldn't need to. For example, why would a WorkItem need a `caseId` when it's going to be attached to a document that is attached to the case? Also, why do entities reference each other by `id` rather than using the reference itself?

File: web-client/src/presenter/sequences/gotoDocumentDetailSequence.js (hunk 2 above)
Comment (codyseibert) [reply web-client/src/presenter/sequences/gotoDocumentDetailSequence.js:4]: that was by accident lol

File: web-client/src/views/WorkQueue.jsx (hunk 4 above)
Comment (codyseibert) [reply web-client/src/views/WorkQueue.jsx]: probably not.. it should always be set I think

**Comment by codyseibert** [comment]
I think there are a couple of attributes that we don't need to be stored on the entities because the parent entity knows about them.  The main issue is dynamo makes it hard to know about the parent entity in certain situations, so it is easier to just also store the id to the parent in the child.

For example, we need the ability to query for all workitems attached to a user.  To achieve this, we have to basically store the workitem twice.  Once in the case json, and as a standalone record.  When it's a syandalone record, we have no way to know what document or case it belongs to.

**Comment by dknesek** [comment]
> I think there are a couple of attributes that we don't need to be stored on the entities because the parent entity knows about them. The main issue is dynamo makes it hard to know about the parent entity in certain situations, so it is easier to just also store the id to the parent in the child.
> 
> For example, we need the ability to query for all workitems attached to a user. To achieve this, we have to basically store the workitem twice. Once in the case json, and as a standalone record. When it's a syandalone record, we have no way to know what document or case it belongs to.

@codyseibert My initial thought is that the structure of an entity graph and the data model in dynamo are two different things. The design of entities shouldn't be influenced by our choice of DB. Some object that's part of our persistence gateway (Uncle Bob calls these "data mappers") should know how the object model needs to be converted to our data model. Let's talk about it.

**Pull request merged**